### PR TITLE
Source Hubspot: turn on default HttpAvailabilityStrategy

### PR DESCRIPTION
**Pull request by grubberr**
Signed-off-by: Sergey Chvalyuk <grubberr@gmail.com>

## What
Issue: https://github.com/airbytehq/airbyte/issues/21782 https://github.com/airbytehq/airbyte/issues/22255
Turn on default HttpAvailabilityStrategy - it works ok for this connector

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-hubspot/integration_tests/expected_records.jsonl [hunk 1]
```diff
@@ -21,7 +21,7 @@
...
...ot_owner_id": "52550153", "hubspot_team_id": null, "hubspotscore": null, "industry": "ACCOUNTING", "is_public": null, "lifecyclestage": "lead", "linkedin_company_page": "Test", "linkedinbio": null, "name": "Test", "notes_last_contacted": null, "notes_last_updated": "2023-02-03T07:00:00+00:00", "notes_next_activity_date": null, "num_associated_contacts": 0.0, "num_associated_deals": null, "num_contacted_notes": 0.0, "num_conversion_events": null, "num_conversion_events_cardinality_sum_d095f14b": null, "num_notes": 2.0, "numberofemployees": 1.0, "phone": null, "recent_conversion_date": null, "recent_conversion_date_timestamp_latest_value_72856da1": null, "recent_conversion_event_name": null, "recent_conversion_event_name_timestamp_latest_value_66c820bf": null, "recent_deal_amount": null, "recent_deal_close_date": null, "state": "Test", "timezone": "3", "total_money_raised": null, "total_revenue": null, "twitterbio": null, "twitterfollowers": null, "twitterhandle": null, "type": "PROSPECT", "web_technologies": null, "website": "test.test", "zip": "00000"}, "createdAt": "2023-01-30T23:22:56.969Z", "updatedAt": "2023-02-03T07:03:57.136Z", "archived": false}, "emitted_at": 1675336936473}
```
Comment (erohmensing) [inline airbyte-integrations/connectors/source-hubspot/integration_tests/expected_records.jsonl:24]: Why did this change?

**Review by erohmensing — COMMENTED** [review]
Looks good! Just some questions about things that look less related. Deleting code 🥳

File: airbyte-integrations/connectors/source-hubspot/source_hubspot/source.py [hunk 2]
```diff
@@ -136,51 +131,3 @@ def streams(self, config: Mapping[str, Any]) -> List[Stream]:
...
-        # TODO assert all streams exist in the connector
-        # get the streams once in case the connector needs to make any queries to generate them
-        stream_instances = {s.name: s for s in self.streams(config)}
-        state_manager = ConnectorStateManager(stream_instance_map=stream_instances, state=state)
-        self._stream_to_instance_map = stream_instances
-        with create_timer(self.name) as timer:
-            for configured_stream in catalog.streams:
-                stream_instance = stream_instances.get(configured_stream.stream.name)
-                if not stream_instance and configured_stream.stream.name == "quotes":
-                    logger.warning("Stream `quotes` does not exist in the source. Skip reading `quotes` stream.")
```
[the remark this answers is not in the document]
Comment (grubberr) [reply airbyte-integrations/connectors/source-hubspot/source_hubspot/source.py:161]: yes, this `def read` has to be removed in this PR https://github.com/airbytehq/airbyte/pull/18546

File: airbyte-integrations/connectors/source-hubspot/integration_tests/expected_records.jsonl (hunk 1 above)
Comment (grubberr) [reply airbyte-integrations/connectors/source-hubspot/integration_tests/expected_records.jsonl:24]: After this PR https://github.com/airbytehq/airbyte/pull/21775 this record already changed and CAT tests started to fail for me

**Review by erohmensing — APPROVED** [review]

**Pull request merged**